File: HackerRank Solutions/Algorithms/Implementation/Viral Advertising/Viral Advertising.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, liked, cumulativeLikes, shared;
    cin >> n;
    shared = 6;
    cumulativeLikes = 2;
    for (int i=2; i<=n; i++){
        liked = floor(shared / 2);
        shared = (liked) * 3;
        cumulativeLikes += liked;
    }
    cout << cumulativeLikes << endl;
    return 0;
}

File: HackerRank Solutions/Problem Solving/Algorithms/Implementation/Viral Advertising/Viral Advertising.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the viralAdvertising function below.
def viralAdvertising(n):
    shared = 6
    liked = 0
    cumulativeLikes = 2
    for i in range(2,n+1):
        liked = math.floor(shared/2)
        shared = liked*3
        cumulativeLikes += liked
    return cumulativeLikes
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    result = viralAdvertising(n)

    fptr.write(str(result) + '\n')

    fptr.close()
